app/(main)/league/[leagueId]/entry/[entryId]/week/Week.test.tsx

```diff
@@ -153,22 +166,105 @@ describe('Week', () => {
     mockUseAuthContext.isSignedIn = true;
-    mockCreateWeeklyPicks.mockResolvedValue({});
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: [],
+        eliminated: false,
+      },
+    ]);
 
     render(
       <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
     );
-    await waitFor(() => {
-      expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
-    });
+
+    // Wait for the main content to be displayed
+    await waitFor(
+      () => {
+        expect(screen.getByTestId('weekly-picks')).toBeInTheDocument();
+      },
+      { timeout: 5000 },
+    );
+
+    expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
+  });
+
+  test('should not show previous weeks picks if there is no history of selected teams', async () => {
+    mockUseAuthContext.isSignedIn = true;
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: [],
+        eliminated: false,
+      },
+    ]);
+
+    render(
+      <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
+    );
+
+    await waitFor(
+      () => {
+        expect(screen.getByTestId('weekly-picks')).toBeInTheDocument();
+      },
+      { timeout: 5000 },
+    );
+
+    const userPickHistory = screen.queryByTestId('user-pick-history');
+    expect(userPickHistory).not.toBeInTheDocument();
+  });
+
+  test('should show previous weeks picks if their are picks in selected teams history', async () => {
+    mockUseAuthContext.isSignedIn = true;
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: ['Packers', 'Browns'],
+        eliminated: false,
+      },
+    ]);
+
+    render(
+      <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
+    );
+
+    await waitFor(
+      () => {
+        expect(screen.getByTestId('weekly-picks')).toBeInTheDocument();
+      },
+      { timeout: 5000 },
+    );
+
+    const userPickHistory = screen.getByTestId('user-pick-history');
+    expect(userPickHistory).toBeInTheDocument();
+
+    const userPickHistoryLogos = screen.queryAllByTestId('league-history-logo');
+    expect(userPickHistoryLogos).toHaveLength(2);
+    expect(userPickHistoryLogos[0]).toHaveAttribute(
+      'src',
+      '/_next/image?url=https%3A%2F%2Fexample.com%2Fpackers.png&w=128&q=75',
+    );
+    expect(userPickHistoryLogos[1]).toHaveAttribute(
+      'src',
+      '/_next/image?url=https%3A%2F%2Fexample.com%2Fbrowns.png&w=128&q=75',
+    );
   });
 
   xtest('should show success notification after changing your team pick', async () => {
```

Comment:
After confirming with Shashi that we're using `it`, please change this

Reply:
✅ done.